Tests/UnitTests/Tests/ProxyTests.cpp

```diff
@@ -35,27 +35,45 @@ DEFINE_TEST_CLASS(ProxyTests)
         VERIFY_ARE_EQUAL_STR(L"localhost", proxyName.c_str());
     }
 
-    DEFINE_TEST_CASE(NamedProxyFormattingCopiedLogic)
+    DEFINE_TEST_CASE(NamedProxyExplicitDefaultPort)
     {
-        DEFINE_TEST_CASE_PROPERTIES(NamedProxyFormattingCopiedLogic);
-        // This test intentionally copies the (fixed) logic for building the proxy string to guarantee coverage
-        // without needing to invoke broader WinHTTP setup.
-        xbox::httpclient::Uri uri{"http://127.0.0.1:8888"};
-        auto formatted = WinHttpProvider::BuildNamedProxyString(uri);
-        http_internal_wstring wProxyHost = utf16_from_utf8(uri.Host());
-        VERIFY_ARE_EQUAL_STR(L"127.0.0.1:8888", formatted.c_str());
-
-        // Demonstrate what the buggy pattern would have produced (length 12 vs 14 expected characters including colon?)
-        uint16_t port = uri.Port();
-        http_internal_basic_stringstream<wchar_t> buggy;
-        buggy.imbue(std::locale::classic());
-        // Simulate original insertion that treated uint16_t as wchar_t
-        buggy << wProxyHost << L":" << static_cast<wchar_t>(port);
-        auto buggyStr = buggy.str();
-        // Ensure buggy result differs from correct formatting (regression guard)
-        VERIFY_IS_TRUE(buggyStr != formatted);
-        // The buggy output should have total length host + 1 (char for ':') + 1 (single wide char) == wProxyHost.length() + 2
-        VERIFY_ARE_EQUAL(wProxyHost.length() + 2, buggyStr.length());
+        DEFINE_TEST_CASE_PROPERTIES(NamedProxyExplicitDefaultPort);
+        // Explicit default port (http://example.com:80) is preserved by Uri parsing (Port()==80 not treated as default)
+        xbox::httpclient::Uri uri{"http://example.com:80"};
+        auto proxyName = WinHttpProvider::BuildNamedProxyString(uri);
+        VERIFY_ARE_EQUAL_STR(L"example.com:80", proxyName.c_str());
+
+        // Verify no default port works
+        xbox::httpclient::Uri uri2{"http://example.com"};
+        auto proxyName2 = WinHttpProvider::BuildNamedProxyString(uri2);
+        VERIFY_ARE_EQUAL_STR(L"example.com", proxyName2.c_str());
+    }
+
+    DEFINE_TEST_CASE(NamedProxyIPv6LiteralWithPort)
+    {
+        DEFINE_TEST_CASE_PROPERTIES(NamedProxyIPv6LiteralWithPort);
+        // Uri class should normalize; host for IPv6 literal typically without brackets when accessed via Host()
+        xbox::httpclient::Uri uri{"http://[2001:db8::1]:3128"};
+        auto proxyName = WinHttpProvider::BuildNamedProxyString(uri);
+        // Expect host + :port (no brackets re-added by BuildNamedProxyString)
+        // If Uri::Host() preserves brackets, adjust expected accordingly; we detect by checking first char
+        auto hostUtf16 = utf16_from_utf8(uri.Host());
+        http_internal_wstring expected = hostUtf16;
+        if (!uri.IsPortDefault() && uri.Port() > 0)
+        {
+            expected.push_back(L':');
+            expected.append(std::to_wstring(static_cast<unsigned int>(uri.Port())));
```

Comment:
another unnecessary cast?

```suggestion
            expected.append(std::to_wstring(uri.Port()));
```

Reply:
Yes, removed